pkg/blobinfocache/internal/prioritize/prioritize.go

```diff
@@ -10,11 +10,16 @@ import (
 	"github.com/opencontainers/go-digest"
 )
 
-// replacementAttempts is the number of blob replacement candidates returned by destructivelyPrioritizeReplacementCandidates,
+// replacementAttempts is the number of blob replacement candidates with known location returned by destructivelyPrioritizeReplacementCandidates,
 // and therefore ultimately by types.BlobInfoCache.CandidateLocations.
 // This is a heuristic/guess, and could well use a different value.
 const replacementAttempts = 5
 
+// replacementUnknownLocationAttempts is the number of blob replacement candidates with unknown Location returned by destructivelyPrioritizeReplacementCandidates,
+// and therefore ultimately by types.BlobInfoCache.CandidateLocations.
+// This is a heuristic/guess, and could well use a different value.
+const replacementUnknownLocationAttempts = 3
```

Comment:
Absolutely non-blocking: `2`? That’s enough for one compressed and one uncompressed digest, or two differently-compressed digests, on top of `primaryDigest` that is going to be tried anyway.